### ISSUE TITLE
prepdocs.py: error: argument --openaiservice: expected one argument

### ISSUE THREAD

**Issue by engrharisijaz** [Azure-Samples/azure-search-openai-demo#680]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ ] feature request
- [X] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
> azd auth login
> azd init -t azure-search-openai-demo
> azd env set OPENAI_HOST openai
> azd env set OPENAI_ORGANIZATION Linkers
> azd env set OPENAI_API_KEY sk-9B1oh0A8f7o2*****************
> azd up

### Any log messages given by the failure
> Running "prepdocs.py"
usage: prepdocs.py [-h] [--datalakestorageaccount DATALAKESTORAGEACCOUNT] [--datalakefilesystem DATALAKEFILESYSTEM]
                   [--datalakepath DATALAKEPATH] [--datalakekey DATALAKEKEY] [--useacls] [--category CATEGORY]
                   [--skipblobs] [--storageaccount STORAGEACCOUNT] [--container CONTAINER] [--storagekey STORAGEKEY]
                   [--tenantid TENANTID] [--searchservice SEARCHSERVICE] [--index INDEX] [--searchkey SEARCHKEY]
                   [--openaihost OPENAIHOST] [--openaiservice OPENAISERVICE] [--openaideployment OPENAIDEPLOYMENT]
                   [--openaimodelname OPENAIMODELNAME] [--novectors] [--disablebatchvectors] [--openaikey OPENAIKEY]
                   [--openaiorg OPENAIORG] [--remove] [--removeall] [--localpdfparser]
                   [--formrecognizerservice FORMRECOGNIZERSERVICE] [--formrecognizerkey FORMRECOGNIZERKEY] [--verbose]
                   [files]
prepdocs.py: error: argument --openaiservice: expected one argument

Deploying services (azd deploy)

  (✓) Done: Deploying service backend
  - Endpoint: https://app-backend-slf3sgq5zrshq.azurewebsites.net/

![image](https://github.com/Azure-Samples/azure-search-openai-demo/assets/92294452/d4c99a1f-e5dd-4a2e-a2f4-6ff4e9fbdd72)

### OS and Version?
> Windows 11

### azd version?
> azd version 1.3.1 (commit b5030da0f29ffe98664c40450187fd8a4cb0d157)

### Mention any other details that might be useful

> I am using non-Azure Open AI instance. The app get deployed by did not generate any response in chat or ask a question mode. Ask a questions also give an Error "No such organization: Linkers. "
![image](https://github.com/Azure-Samples/azure-search-openai-demo/assets/92294452/871dd386-143f-472c-b626-a4571df5dfc7)

**Comment by engrharisijaz** [comment]
[Another error occured](prepdocs.py: error: argument --openaideployment: expected one argument)

So, similarly I have changed

--openaideployment $env:AZURE_OPENAI_EMB_DEPLOYMENT
 
to:

--openaideployment `"$env:AZURE_OPENAI_EMB_DEPLOYMENT`"

But still Error: No such organization: Linkers.
![image](https://github.com/Azure-Samples/azure-search-openai-demo/assets/92294452/f483a923-24bf-4c04-ad4e-de40b8bda41d)

**Comment by pamelafox** [comment]
cc @carlotta94c since this is related to non-Azure OpenAI deployment.

Can you set logging mode to debug in prepdocs.py?

logging.basicConfig(logging.DEBUG)

That will then show you the URL that OpenAI is trying to access, and you can compare that URL to what you expect it to be (perhaps in the OpenAI dashboard?). I don't have a non-Azure OpenAI account yet for replication.

**Comment by gramster** [comment]
> 
> `` --openaideployment `"$env:AZURE_OPENAI_EMB_DEPLOYMENT`" ``
> 

Did you misstype that? As the quotes are mismatched.

**Comment by pamelafox** [comment]
No, that's apparently how Powershell escapes quotes? https://www.delftstack.com/howto/powershell/powershell-escape-single-and-double-quote/

**Comment by pamelafox** [comment]
@engrharisijaz I haven't seen the error "No such organization" before but I am assuming it comes from the OpenAI server not recognizing your organization. You can also print or log out what's being set on this line:

            openai.organization = args.openaiorg

And make sure it matches what's in your OpenAI account.
